partner_firstname/models/res_partner.py

```diff
@@ -127,6 +127,11 @@ def _get_whitespace_cleaned_name(self, name, comma=False):
 
         Removes leading, trailing and duplicated whitespace.
         """
+        if isinstance(name, bytes):
```

Comment:
Is this module the right place to solve this problem? It should not be possible to set a name on a partner to bytes in the first place I think...

Reply:
Standard Odoo automatically converts the vals from bytes to strings while creating/writing Char fields. This is done [here](https://github.com/odoo/odoo/blob/14.0/odoo/tools/pycompat.py#L35-L36) when invoking `odoo.tools.pycompat.to_text` to converts bytes to string.

In this particular case, method `_get_whitespace_cleaned_name()` is executed on the vals before the bytes are converted to strings, which is done later when invoking the `super()`. So I think this is the right place to solve this problem, unless someone else wants to refactor the module itself.